apis/v1alpha1/cluster_fentry_program_types.go

```diff
@@ -31,16 +30,26 @@ type ClFentryProgramInfo struct {
 // ClFentryLoadInfo contains the program-specific load information for Fentry
 // programs
 type ClFentryLoadInfo struct {
-	// Function is the name of the function to attach the Fentry program to.
+	// function is the name of the function to attach the Fentry program to.
+	// +kubebuilder:validation:Pattern="^[a-zA-Z][a-zA-Z0-9_]+."
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=64
 	Function string `json:"function"`
 }
 
+type AttachTypeAttach string
+
+const (
+	Attach  AttachTypeAttach = "Attach"
+	Dettach AttachTypeAttach = "Detach"
```

Comment:
I was not intending to support these semantics.  For all program types, if an entry is present in the list, the intent is for the link to be attached.  If an entry is not present in the list, the intent is for it not to be attached.  So, the only valid attach type would be `Attach`.

Reply:
Per my comment on the review pr, I was trying to make both the syntax and semantics the same for links for all program types: present == attach, and not present == detach.

Reply:
as u saw in the review they want us to move away from using people and replace with enum it might help down the road for not introducing breaking changes if we need more state